collect/cache/traceSentCache.go

```diff
@@ -0,0 +1,24 @@
+package cache
+
+import (
+	"github.com/honeycombio/refinery/types"
+)
+
+type TraceSentRecord interface {
+	// Kept returns whether the trace was kept (sampled and sent to honeycomb) or dropped.
+	Kept() bool
+	// Rate() returns the sample rate for the trace
+	Rate() uint
+	// DescendantCount returns the count of items associated with the trace, including all types of children like span links and span events.
+	DescendantCount() uint
+	// Count records additional spans in the totals
+	Count(*types.Span)
+}
+
+type TraceSentCache interface {
+	// Record preserves the record of a trace being sent or not.
+	Record(trace *types.Trace, keep bool)
+	// Check tests if a trace corresponding to the span is in the cache; if found, it returns the appropriate TraceSentRecord and true,
+	// else nil and false.
+	Check(span *types.Span) (TraceSentRecord, bool)
```

Comment:
How do you feel about using a TraceID instead of a span. Something like:

```suggestion
	Get(traceID string) (TraceSentRecord, bool)
```

Reply:
I did that originally, but this is partly prep for a future change where we keep more information in the kept cache regarding not just total number of descendants, but also keep separate counts of other things like span events, links, etc. To do that I need to inspect the whole span.